lib/stdlib/src/queue.erl

```diff
@@ -409,9 +409,8 @@ filter_r(Fun, [X|R0]) ->
 -spec filtermap(Fun, Q1) -> Q2 when
       Fun :: fun((Item) -> boolean() | {'true', Value}),
       Q1 :: queue(Item),
-      Q2 :: queue(Item | Value),
-      Item :: term(),
-      Value :: term().
+      Q2 :: queue(Item),
+      Value :: Item.
```

Comment:
```suggestion
      Value :: term().
```
Resulting values don't necessarily have to be of the same type as the items in the queue.

Reply:
I have fixed.